static/code/svg.js

```diff
@@ -107,5 +117,144 @@ svgEditor.getString = function() {
+      var styleCloseAction = document.getElementById('close-action').style;
+      var styleOpenAction = document.getElementById('open-action').style;
+      if (svgEditor.canvas.pathActions.closed_subpath) {
+        styleCloseAction.display = 'none';
+        styleOpenAction.display = 'block';
+      } else {
+        styleOpenAction.display = 'none';
+        styleCloseAction.display = 'block';
+      }
+    } else {
+      stylePathEdit.display = 'none';
+      styleSelect.display = 'block';
+      styleNodeActions.display = 'none';
+    }
+    var button = document.getElementById('mode-' +
+                                         svgEditor.updateToolbox.oldMode_);
+    if (button) {
+      button.classList.remove('selected');
+    }
+    svgEditor.updateToolbox.oldMode_ = mode;
+    button = document.getElementById('mode-' + mode);
+    if (button) {
+      button.classList.add('selected');
+    }
+  }
+
+  // Show or hide action buttons that apply to one or more selected elements.
+  var selected = svgEditor.canvas.getSelectedElems();
+  if (selected.length !== svgEditor.updateToolbox.oldSelectedCount_) {
+    var actions = document.getElementById('selected-actions');
+    var singleActions = document.getElementById('selected-single-actions');
+    actions.style.display = selected.length > 0 ? 'block' : 'none';
+    singleActions.style.display = selected.length === 1 ? 'block' : 'none';
+    svgEditor.updateToolbox.oldSelectedCount_ = selected.length;
+  }
+  if (selected.length === 1) {
+    var element = selected[0];
+    document.getElementById('convertpath-action').style.display =
+        element.tagName == 'path' ? 'none' : 'block';
+
+    var fillStroke = svgEditor.getFillStroke(element);
+    var fill = fillStroke[0];
+    var stroke = fillStroke[1];
+
+    var fillRect = document.getElementById('fillRect');
+    var fillNoneStyle = document.getElementById('fillNone').style;
+    if (fill === 'fillNone') {
+      fillNoneStyle.display = 'inline';
+      fillRect.setAttribute('class', 'fillWhite');
+    } else {
+      fillNoneStyle.display = 'none';
+      fillRect.setAttribute('class', fill);
+    }
+
+    var strokeRect = document.getElementById('strokeRect');
+    if (stroke === 'strokeNone') {
+      strokeRect.setAttribute('stroke-width', '1');
+      strokeRect.setAttribute('stroke', '#d40000');
+      strokeRect.setAttribute('class', '');
+    } else {
+      strokeRect.setAttribute('stroke-width', '4');
+      strokeRect.setAttribute('stroke', '');
+      strokeRect.setAttribute('class', stroke);
+    }
+  }
+};
+
+svgEditor.updateToolbox.oldMode_ = '';
+svgEditor.updateToolbox.oldSelectedCount_ = 0;
+
+/**
+ * Find the fill and stroke for an element.
+ * @param {!Element} element Element with style.
+ * @return {!Array<string>} Array with normalized fill and stroke strings.
+ */
+svgEditor.getFillStroke = function(element) {
+  if (!element) throw TypeError('Element not provided');
```

Comment:
Are you concerned the compiler might not enforce the type declaration for `element`?